test/Test/Data/StrMap.purs

```diff
@@ -34,7 +34,8 @@ instance showInstruction :: (Show k, Show v) => Show (Instruction k v) where
 instance arbInstruction :: (Arbitrary v) => Arbitrary (Instruction String v) where
   arbitrary = do
     b <- arbitrary
-    k <- arbitrary
+    kIshasOwnProperty <- (&&) <$> arbitrary <*> arbitrary
+    k <- if kIshasOwnProperty then pure "hasOwnProperty" else arbitrary
```

Comment:
Might `frequency` be a more idiomatic way of saying that we should occasionally use 'hasOwnProperty' as a key? https://pursuit.purescript.org/packages/purescript-quickcheck/3.1.1/docs/Test.QuickCheck.Gen#v:frequency